File: Padrões Gof/Estruturais/Adapter/Adapter01.py
# Interface do Cliente
class ITarget:
    def request(self):
        pass

# Cliente
class Client:
    def __init__(self, target):
        self.target = target
    

    def makeRequest(self):
        print("Fazendo uma Requisição")
        self.target.request()



# Serviço Existente (Adaptee)
class Adaptee:
    def specificRequest(self):
        print("Requisição Especifica do Adaptee")



# Adaptador (Adapter)
class Adapter(ITarget):
    def __init__(self, adaptee):
        self.adaptee = adaptee


    def request(self):
        self.adaptee.specificRequest()



# Utilizando o Adaptador #######################################
adaptee = Adaptee()
adapter = Adapter(adaptee)
client = Client(adapter)

client.makeRequest()

File: Padrões Gof/Estruturais/Adapter/Adapter01.js
// interface do cliente
 class ITarget{
   request(){}
 } 
// cliente
class Client{
  constructor(target){
    this.target = target;
  }
  makeRequest(){
    console.log("Fazendo uma Requisição");
    this.target.request();
  }
}
// Serviço Existente(Adapter)
class Adaptee{
  specificRequest(){
    console.log("Requisição Especifica do Adaptee");
  }
}
// Adaptador(Adapter)
class Adapter{
  constructor(adaptee){
    this.adaptee = adaptee;
  }
  request(){
    this.adaptee.specificRequest();
  }
}
//Utilizando o Adaptador
const adaptee = new Adaptee();
const adapter = new Adapter(adaptee);
const client = new Client(adapter);

client.makeResquest();